### PR TITLE
tiscamera: Init at 0.9.1

### PR DESCRIPTION
**Pull request by jraygauthier**
###### Motivation for this change

Linux uvc drivers for the imaging source camera. 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by jraygauthier** [comment]
The uvc dynamic controls are registered through #41737

File: pkgs/os-specific/linux/tiscamera/default.nix [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tiscamera";
+  version = "0.9.1";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "TheImagingSource";
+    repo = "${pname}";
```
Comment (ryantm) [inline pkgs/os-specific/linux/tiscamera/default.nix]: repo = pname;

File: pkgs/os-specific/linux/tiscamera/default.nix [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    rev = "v-${pname}-${version}";
```
Comment (ryantm) [inline pkgs/os-specific/linux/tiscamera/default.nix]: rev = "v-${name}";

File: pkgs/os-specific/linux/tiscamera/default.nix [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  # dependency on `libtcam` (which itself is built as part of this build). In order to allow
+  # that, we set the dynamic linker's path to point on the build time location of the library.
+  preBuild = ''
+    export LD_LIBRARY_PATH=$PWD/src:$LD_LIBRARY_PATH
+  '';
+
+  meta = with lib; {
+    description = "The Linux sources and UVC firmwares for The Imaging Source cameras";
+    homepage = https://github.com/TheImagingSource/tiscamera;
+    licenses = with licenses; [ asl20 ];
```
Comment (ryantm) [inline pkgs/os-specific/linux/tiscamera/default.nix]: `license = with licenses; [ asl20 ];`
fixes eval.

**Review by ryantm — CHANGES_REQUESTED** [review]

**Comment by jraygauthier** [comment]
@ryantm : Done.

**Comment by xeji** [comment]
@GrahamcOfBorg build tiscamera

**Pull request merged**